### DISCUSSION TITLE
I cannot update my SDK extension in visual studio (Stuck on 1.423*)

### DISCUSSION THREAD

**Discussion by AlexanderBlackman** [microsoft/WindowsAppSDK#4278] [Q&A] [answered]
The extension hasn't  updated from the September version.
![extension](https://github.com/microsoft/WindowsAppSDK/assets/51733876/32884d3d-cd79-432e-9a76-48ff8e15068c)
I've tried uninstalling the extension, but it returns unupdated.
I've tried uninstalling and reinstalling the workload in VS Installer, but it returns unupdated. 
![Addon](https://github.com/microsoft/WindowsAppSDK/assets/51733876/de500902-7ee0-45be-a98f-b2e35146309e)

I've tried installing  [windowsappruntimeinstall-x64](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/downloads) but nothing changed.

What should I do?  Reformat my entire computer and reinstall everything from scratch?  There must be a better way but I really need access to the new WinUI friendly test templates and updates.

**Comment by DarranRowe** [comment, marked as answer]
AFAIK, they won't be updated until 17.10. Currently, the templates are distributed as part of Visual Studio itself, and Visual Studio has some rules as to when components are updated.

I do remember some mention of them figuring out some other means of updating the templates, but right now they will be stuck until at least 17.10. Unless there is a set of third party templates available, your best option is to use the [templates in the repository](https://github.com/microsoft/WindowsAppSDK/tree/main/dev/VSIX/ProjectTemplates) as a source for editing and updating any newly created project.

**Reply by AlexanderBlackman** [reply]
That's frustrating, but thank you for your answer.
